tests/integration/test_integration.py

```diff
@@ -97,8 +100,8 @@ async def wait_for_certificate_to_be_provided(ops_test: OpsTest) -> None:
 
 
 @pytest.fixture(scope="module")
-async def deploy_vault(ops_test: OpsTest, request) -> None:
-    """Build the charm-under-test and deploy it."""
+async def deployed_vault(ops_test: OpsTest, request) -> None:
```

Comment:
Why was the name change necessary? It is actually deploying Vault.

Reply:
I don't think fixtures should be used as actions, they should represent states. We're not deploying a vault for every test that includes this fixture, we're effectively asserting that vault has been deployed (although this and other fixtures could be improved a lot -- that's something I would like to tackle independently).

It might seem nitpicky and unnecessary in this case, but it became an issue when you have a fixture called `initialize_vault_leader` (also renamed). I wanted a function that did exactly that, so what do I call it? They start to conflict, and the fixtures names don't accurately represent what they are meant to be used for. I just renamed both of them to keep them consistent.